marienfressinaud/FreshRSS
# UI: Add optional thumbnail and summary on feed items

Closes #561

**Changes proposed in this pull request**:

Implements Thumbnails as well as Summary on feed items in the normal view.
Example feed: `https://www.futura-sciences.com/rss/definitions.xml`

![image](https://user-images.githubusercontent.com/1128206/130869429-f9e70a0d-5d05-43ab-884c-7e00e444fc20.png)

Settings are implemented under _Configuration > Display_ and saved in per-user settings:

![image](https://user-images.githubusercontent.com/1128206/130869513-ff2cc9ae-63c0-48bb-9382-e589440c2db9.png)

When enabling Thumbnails only, the image is smaller:

![image](https://user-images.githubusercontent.com/1128206/130869668-30358195-bdbc-477b-85d5-f92049cd3392.png)

I've tested the changes with all built-in themes and made specific adjustments for the Compact theme:

![image](https://user-images.githubusercontent.com/1128206/130869816-5525dc2d-0b2e-47e7-8b3e-f88e045f9d45.png)

The right-to-left variant is also handled:

![image](https://user-images.githubusercontent.com/1128206/130869974-5f586bc2-5531-47d0-a5ee-b16b7b9f3e83.png)

**How to test the feature manually:**

1. Under _Configuration > Display_, check _Thumbnail_, _Summary_, or both
2. Submit changes and go back to the normal view
3. New elements are shown in the feed

**Pull request checklist:**

- [ ] unit tests written (sorry, did not implement tests but let me know if/what tests are needed)
- [ ] documentation updated (should not be needed but let me know if I need to change something)

**Additional notes:**

Regarding translation, items are translated in English and French only at the moment, and left in English for other locales.

I encountered a conflicting `position: absolute` CSS rule from [this commit](https://github.com/FreshRSS/FreshRSS/commit/c5274cfa298f1020bab30bd6026323be9e4be43d#diff-65f6b443253a1121cd18180e99827a32b608e5c3a2fe42efb6f5a6317e3e8d1eR659) that cause glitches when hovering items having a thumbnail or summary, so using an additional CSS class on the `title` item when either element is enabled, the conflicting rule is overridden only when needed and left as is otherwise to avoid reintroducing #367 accidentally.

As this is my first pull request on this project, I'm not very familiar with the codebase and may have missed something. Please let me know if anything is missing so I can improve. 🙂
